credentials/fetcher_default.go

```diff
@@ -112,12 +120,12 @@ func (s *FetcherDefault) fetchParallel(ctx context.Context, locations []url.URL)
 		}
 	}()
 
-	go s.resolveAll(done, errs, locations)
+	go s.resolveAll(ctx, done, errs, locations)
 
 	select {
-	case <-ctx.Done():
-		s.l.WithError(ctx.Err()).Errorf("Ignoring JSON Web Keys from at least one URI because the request timed out waiting for a response.")
-		return ctx.Err()
+	case <-time.After(s.cancelAfter):
+		s.l.Errorf("Ignoring JSON Web Keys from at least one URI because the request timed out waiting for a response.")
+		return context.DeadlineExceeded
 	case <-done:
```

Comment:
ctx.Done is now missing, no? The parent of `fetchParallel` might still cancel the context.

Reply:
fixed